src/__mocks__/eslint.js

```diff
@@ -18,7 +18,8 @@ module.exports = Object.assign(eslint, {
 
 function MockCLIEngine(...args) {
   global.__PRETTIER_ESLINT_TEST_STATE__.eslintPath = __filename
-  CLIEngine.apply(this, args)
+  // eslint-disable-next-line
+  CLIEngine.apply(this, args);
   // not sure why, but in some cases, this.executeOnText is undefined...
   // so we create a fakeCLIEngine to get a copy of that function
   // and call it with apply :)
```

Comment:
ESLint seems to be getting angry about the next line...

Reply:
MockCLIEngine is invoked with `new`, but CLIEngine isn't. Maybe this would work?

```js
return new (Function.prototype.bind.apply(CLIEngine, args));
```

Reply:
Thanks! I'll try it!

Reply:
Hmm. Didn't work :-(

Reply:
I was able to bump the version of ESLINT to latest in my branch and prettier-eslint.  I'll open a pull request and see your thoughts.

Reply:
whoops ignore this lol wrong repo :(

Reply:
How about this?

```js
function MockCLIEngine(...args) {
  global.__PRETTIER_ESLINT_TEST_STATE__.eslintPath = __filename

  // run the original constructor function on `this`
  new (Function.prototype.bind.apply(CLIEngine, [this, ...args]))

  const fakeCLIEngine = new CLIEngine(...args)
  this.getConfigForFile = mockGetConfigForFileSpy
  this._originalExecuteOnText = fakeCLIEngine.executeOnText.bind(fakeCLIEngine)
  this.executeOnText = mockExecuteOnTextSpy
}

```

_Edit_: turns out the `fakeCLIEngine` is needed so `_originalExecuteOnText` has a "real" instance to run against (hence the `.bind`). No more errors.

Reply:
Thanks for your help friends! I got it worked out by forgetting `this` and just using `fakeCLIEngine` :)